### PR TITLE
Introduce MinDistance to AreaBeam projectile

### PR DESCRIPTION
**Pull request by PunkPun**
Superseding #20170

### CODE REVIEW

**Comment by PunkPun** [comment]
TS disrupters only shot as far as they aimed, so no need to adjust them

File: OpenRA.Mods.Common/Projectiles/AreaBeam.cs [hunk 1]
```diff
@@ -136,7 +139,19 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
...
 			var dir = new WVec(0, -1024, 0).Rotate(WRot.FromYaw(towardsTargetFacing));
-			target += dir * info.BeyondTargetRange.Length / 1024;
+			var dist = (args.SourceActor.CenterPosition - target).Length;
+			int extraDist;
+			if (info.MinDistance.Length > dist)
+			{
+				if (info.MinDistance.Length - dist < info.BeyondTargetRange.Length)
+					extraDist = info.BeyondTargetRange.Length;
+				else
+					extraDist = info.MinDistance.Length;
```
Comment (penev92) [inline OpenRA.Mods.Common/Projectiles/AreaBeam.cs]: Won't this add MinDistancw behind the target? Shouldn't it be MinDistance - dist?

Comment (PunkPun) [reply OpenRA.Mods.Common/Projectiles/AreaBeam.cs]: yes mb, it should be `MinDistance - dist`

**Review by AspectInteractive2 — APPROVED** [review]
Code looks good, the logic is correct and is well implemented.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/96d92645f486670797a4845cfc2d2e42a3b438ce...0f4f64ccfa2dd1312a776fb45f13b1540bb6b46b)

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/e7a37d6b6417c2204e7c118b3e4b354ac91949e4...25f01794b0bbdbaef124c8e3f5ce43e669c070e9)

**Comment by abcdefg30** [comment]
For reference, the prep commit: https://github.com/OpenRA/OpenRA/commit/c01e4043e819897f30b5e2c6d6044556cad3491a

**Pull request merged**